util/fibers/fiber2.h

```diff
@@ -70,6 +70,9 @@ class Fiber {
 
   void Join();
 
+  // Join fiber if it's running, else do nothing.
+  void PermJoin();
```

Comment:
Reading this, I wouldn't know `Perm` stands for "permissive" (I woulda thought about permanent maybe?)
How about `JoinIfJoinable()`?
Anyway, looks like a good utility :)

Reply:
```suggestion
  void JoinIfNeeded();
```